server/src/test/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapperTests.java

```diff
@@ -2021,23 +2021,25 @@ protected boolean supportsEmptyInputArray() {
     }
 
     private static class DenseVectorSyntheticSourceSupport implements SyntheticSourceSupport {
-        private final int dims = between(5, 1000);
-        private final ElementType elementType = randomFrom(ElementType.BYTE, ElementType.FLOAT);
+//        private final int dims = between(5, 1000);
+        private final int dims = between(5, 512);
+        private final ElementType elementType = randomFrom(ElementType.BYTE, ElementType.BIT, ElementType.FLOAT);
```

Comment:
@benwtrent 
Thank you for your patience. I have added tests for the BIT type in DenseVectorSyntheticSourceSupport. However, I seem to have discovered another bug. During the parsing process, since it is done sequentially, the `element_type` is not yet parsed when `dims` is being processed. As a result, the default value is FLOAT instead of BIT, causing the `dims` range to be [1, 4096] instead of [8, 32768].
![debug](https://github.com/user-attachments/assets/c8804224-8339-45ec-9b72-ea5e61492f3b)

Here is a test to illustrate the issue:

```
PUT test
{
  "mappings": {
    "properties": {
      "vector": {
        "type": "dense_vector",
        "dims": 8000,
        "index": false,
        "element_type": "bit"
      }
    }
  }
}

{
  "error": {
    "root_cause": [
      {
        "type": "mapper_parsing_exception",
        "reason": "The number of dimensions for field [vector] should be in the range [1, 4096] but was [8000]"
      }
    ],
    "type": "mapper_parsing_exception",
    "reason": "Failed to parse mapping: The number of dimensions for field [vector] should be in the range [1, 4096] but was [8000]",
    "caused_by": {
      "type": "mapper_parsing_exception",
      "reason": "The number of dimensions for field [vector] should be in the range [1, 4096] but was [8000]"
    }
  },
  "status": 400
}
```

This behavior contradicts the expected 32768 dimensions for the BIT type.

To avoid test failures, I have temporarily limited the `dims` in the DenseVectorSyntheticSourceSupport tests to 512 (4096 / 8) when using the BIT type. 

I'm not sure how to proceed from here. Should we address this issue in the current PR, or handle it separately? Any advice would be appreciated.

Reply:
This doesn't have to do with the parsing order. For `bit` vectors, we parse the `dims` as the `number of bits`. Meaning, for `bit`, it must be divisible by `8`, and of course, can be much higher than 4096 (as large as 32768). 

Note from our docs:
>  NOTE: when using bit, the number of dimensions must be a multiple of 8 and must represent the number of bits.

Consequently, your solution of simply `*8` when `bit` makes sense to me :)

Reply:
I apologize if my previous message was unclear. While adding the test code, I found that even when setting the type to `bit`, the `dims` cannot exceed 4096. If you manually execute a REST request with `dims` set above 4096 for a `bit` type, you'll encounter an error stating: "The number of dimensions for field [vector] should be in the range [1, 4096] but was [8000]." If you uncomment line 2024, you'll encounter the same error.

Reply:
Let me try to reproduce.